src/kvstore/NebulaStore.cpp

```diff
@@ -427,6 +427,24 @@ void NebulaStore::removeSpace(GraphSpaceID spaceId, bool isListener) {
   }
 }
 
+nebula::cpp2::ErrorCode NebulaStore::clearSpace(GraphSpaceID spaceId) {
+  folly::RWSpinLock::WriteHolder wh(&lock_);
```

Comment:
The `lock_` is not for DML, it is used for partition related modify, e.g. when add space or drop part.

Maybe you could find all kv_engine by NebulaStore::space, and get all parts by `NebulaStore::allParts` from kv_engine.